lib/triggers/gitHubWebHookTrigger.js

```diff
@@ -0,0 +1,177 @@
+showDetails = string - Show details of actions. IE the actual comment text added if true, otherwise just the username/place of the comment.
+*/
+
+var GitHubWebHookTrigger = function() {
+	GitHubWebHookTrigger.super_.apply(this, arguments);
+};
+
+util.inherits(GitHubWebHookTrigger, BaseTrigger);
+
+var type = "GitHubWebHookTrigger";
+exports.triggerType = type;
+exports.create = function(name, chatBot, options) {
+	var trigger = new GitHubWebHookTrigger(type, name, chatBot, options);
+	trigger.options.path = options.path || '/GitHubWebHook';
+	trigger.options.showDetails = options.showDetails || true;
+	trigger.db = (function(){try{return JSON.parse(fs.readFileSync(trigger.options.dbFile));}catch(err){return {}}})();
+	return trigger;
+};
+
+GitHubWebHookTrigger.prototype._onLoad = function() {
+	if(!(this.options.rooms instanceof Array && this.options.rooms.length===1 && typeof this.options.rooms[0] === 'string')) {
+		this.winston.error(this.chatBot.name+" "+this.name+": options.rooms needs to be a single-length array");
+		return false;
+	}
+	if(!this.options.secret) {
+		this.winston.error(this.chatBot.name+" "+this.name+": options.secret is required");
+		return false;
+	}
+	var that = this;
+	var sm = function(text){that._sendMessageAfterDelay(that.options.room,text);} //...yes, I'm lazy
+	var sd = this.options.showDetails; //I should really start using NPP or something else that has autocompletion.
+	var handler = createHandler({path: '/',secret:this.options.secret});
+	this.chatBot.router.use(this.options.path, function(req, res) {
+		handler(req, res, function(err) {
+			console.log("github webhook incoming");
+			res.statusCode = 404;
+			res.end('no such location')
+		});
+	});
+	handler.on('error', function(err) {
+		that.winston.error(that.name+": Error: ");
+		that.winston.error(err);
+	});
+	handler.on('commit_comment', function (event) {
+		var c = event.payload.comment;
+		var message = "GitHub: "+c.user.login+" has commented on "+c['commit_id'].substr(0,8);
+		message += c.html_url.split(c.commit_id).join(c.commit_id.substr(0,10)); //cut the hash, it's too %*(&@# long.
+		message += that.options.showDetails ? "\n"+c.body : "";
+		sm(message);
+	});
+	handler.on('create', function (event) {
+		var p = event.payload;
+		sm("GitHub: "+p.sender.login+" has created a new "+p['ref_type']+", "+p.ref+", in "+(sd?"https://github.com/":"")+p.repository['full_name']);
+	});
+	handler.on('delete', function (event) {
+		var p = event.payload;
+		sm("GitHub: "+p.sender.login+" has deleted "+p['ref_type']+", "+p.ref+", from "+(sd?"https://github.com/":"")+p.repository['full_name']);
+	});
+//	handler.on('deployment', function (event) {       });  //wtf is this?
+//	handler.on('deployment_status', function (event) {	}); //ditto
+	handler.on('fork', function (event) {
+		var p = event.payload;
+		sm("GitHub: "+p.repository['full_name']+" has been forked to "+(sd?p.html_url:p.forkee['full_name']));
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>['full_name'] is better written in dot notation.<br>Identifier 'html_url' is not in camel case.